server/controller/src/main/java/ai/starwhale/mlops/datastore/DataStore.java

```diff
@@ -610,13 +622,9 @@ private void clearWalLogFiles() {
             if (table == null) {
                 continue;
             }
-            table.lock();
-            try {
-                if (table.getFirstWalLogId() >= 0 && table.getFirstWalLogId() < minWalLogIdToRetain) {
-                    minWalLogIdToRetain = table.getFirstWalLogId();
-                }
-            } finally {
-                table.unlock();
+            var walLogId = table.getFirstWalLogId();
```

Comment:
This id may be -1, but the `minWalLogIdToRetain` has changed.

1. mem-table.update.wal-manager.append // wal-manager.max-id changed
2. dump.clear-wal-log.get-wal-max-id and check table.first-wal-id // table.first-wal-id may be -1
3. mem-table.update first-wal-id

Reply:
1. in this case, the new id will be greater than or equal to the initial minWalLogIdToRetain. Because dumping runs periodically, it is harmless to have a smaller minWalLogIdToRetain.
2. dumping and clearing are always run in the same thread. clearWalLogFiles is always invoked after saveTables. So, this case will never happen.
3. The same as the first case.